oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/model/RunResources.kt

```diff
@@ -0,0 +1,20 @@
+package com.empowerops.volition.ref_oasis.model
+
+import com.empowerops.volition.ref_oasis.model.Helpers.Companion.NullUUID
+import com.empowerops.volition.ref_oasis.optimizer.RunStateMachine
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.channels.Channel
+import java.util.*
+
+class RunResources {
+    val stateMachine: RunStateMachine = RunStateMachine()
+    var runID: UUID = NullUUID
+    var resumeSignal: CompletableDeferred<Unit>? = null
```

Comment:
this is a bad sign. `var something = CompletableDeferred` is almost always a mistake. You almost certainly want a `Channel` here, else you end up writing stateful-asynchronous code, which is the foundation for heisenbug nightmares. 

You really want to tie the existence of `CompletableFuture` to the life cycle of the object. In other words, call `new Model()` then use the completable future on that model _once_, then throw away the model, and make another one when the cercomstance needs it.

Reply:
had the same thought after a while, shared mutable resource is evil and they are kind of slap on when I start to look them differently.